library/src/main/java/com/novoda/downloadmanager/lib/DownloadTask.java

```diff
@@ -449,6 +453,12 @@ private void checkPausedOrCanceled() throws StopRequestException {
         }
     }
 
+    private void checkClientRules() throws StopRequestException {
+        if (!downloadReadyChecker.clientAllowsToDownload(originalDownloadBatch)) {
```

Comment:
The problem is that the batch being sent to the client is incorrect, it's total size and current size are the original values instead of the current ones.

the "easy" fix is to requery for the batch before checking it but the correct fix would be to avoid doing this check like we mentioned and relying on a `REASON` column!